src/figuresClasses.ts

```diff
@@ -1,19 +1,101 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.correctValuesOfFigure();
+  }
+
+  correctValuesOfFigure(): void {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Resulting value should be even number');
+    }
+
+    let sides: number[] = [this.a, this.b, this.c];
+
+    sides = sides.sort((a: number, b: number) => b - a);
 
+    const longestSide: number = sides[0];
```

Comment:
don't add type for primitive js types
```suggestion
    const longestSide = sides[0];
```